### DISCUSSION TITLE
Would it be possible to include code samples for each component demo in the zag.js documentation?

### DISCUSSION THREAD

**Discussion by huangsheng7271** [chakra-ui/zag#1650] [Ideas]
It would be incredibly helpful if the zag.js documentation could include complete code examples for each demo, similar to the approach taken by react-aria. For instance, for the demo shown in the image, providing both the zag.js implementation code and the corresponding CSS would give users a comprehensive understanding of how to recreate the component. This approach would make it much easier for developers to understand and implement zag.js components in their own projects.
<img width="814" alt="截屏2024-07-05 20 43 41" src="https://github.com/chakra-ui/zag/assets/31662714/e8b24a7f-0bef-4f83-921a-8467c7e946e3">

**Comment by segunadebayo** [comment]
Thanks for bringing this up @huangsheng7271

That's definitely something we need to do and @anubra266 is working on

**Reply by timmaier** [reply]
This is definitely something that would be great. Remember struggling in the beginning when starting to use zag.

Radix Vue docs do this really well. Allowing developer to just copy the snippet and have a visually working input out of the box. Then they can tune it to their liking from there.
